### PR TITLE
updated and cleaned PBSPro adapter

### PR DESCRIPTION
**Pull request by mtitov**

### CODE REVIEW

File: src/radical/saga/adaptors/pbspro/pbsprojob.py [hunk 1]
```diff
@@ -161,85 +161,37 @@ def _script_generator(url, logger, jd, ppn, gres, version, is_cray=False,
...
+            stderr_dir = '%s/' % jd.working_directory
+        pbs_params += '#PBS -e %s%s\n' % (stderr_dir, jd.error)
 
     if jd.wall_time_limit:
-        hours = int(jd.wall_time_limit / 60)
-        minutes = jd.wall_time_limit % 60
-        pbs_params += "#PBS -l walltime=%s:%s:00 \n" \
-            % (str(hours), str(minutes))
+        pbs_params += '#PBS -l walltime=%s:%s:00\n' \
+            % (str(int(jd.wall_time_limit / 60)), str(jd.wall_time_limit % 60))
```
Comment (andre-merzky) [inline src/radical/saga/adaptors/pbspro/pbsprojob.py:185]: The `str()` conversion is enforced by `%s` anyway - I would suggest to remove it for clarity.

File: src/radical/saga/adaptors/pbspro/pbsprojob.py [hunk 2]
```diff
@@ -161,85 +161,37 @@ def _script_generator(url, logger, jd, ppn, gres, version, is_cray=False,
...
+
+    if jd.project:
+        pbs_params += '#PBS -A %s\n' % jd.project
+        if 'PBSPro_1' in version:
+            # set both parameters: -P(roject) and -A(accounting)
```
Comment (andre-merzky) [inline src/radical/saga/adaptors/pbspro/pbsprojob.py:190]: Maybe keep that part of the old comment which says that this is a site admin choice in `PBSPro`?

File: src/radical/saga/adaptors/pbspro/pbsprojob.py [hunk 3]
```diff
@@ -251,43 +203,39 @@ def _script_generator(url, logger, jd, ppn, gres, version, is_cray=False,
...
+    if jd.processes_per_host:
+        logger.info('Override detected ppn (%d) with user ppn (%d)' %
+                    (ppn, jd.processes_per_host))
+        ppn = jd.processes_per_host
 
-    if jd.job_contact:
-        pbs_params += "#PBS -m abe \n"
+    if 'cheyenne' in url.host.lower() or \
+       'cheyenne' in os.uname()[1].lower():
+        ppn = 36
```
Comment (andre-merzky) [inline src/radical/saga/adaptors/pbspro/pbsprojob.py:238]: Cheyenne is dead - if you are ok with it, we can remove this switch.

File: src/radical/saga/adaptors/pbspro/pbsprojob.py [hunk 4]
```diff
@@ -298,59 +246,59 @@ def _script_generator(url, logger, jd, ppn, gres, version, is_cray=False,
...
-    elif 'PBSPro_13'       in version: pbs_params += "#PBS -l select=%d\n"          % nnodes                 # pbspro 13
-    elif '4.2.6'           in version: pbs_params += "#PBS -l nodes=%d\n"           % nnodes                 # titan
-    elif '4.2.7'           in version: pbs_params += "#PBS -l mppwidth=%s \n"       % jd.total_cpu_count     # edison, hopper
-    elif 'Version: 5.'     in version: pbs_params += "#PBS -l procs=%d\n"           % jd.total_cpu_count     # torqye 5
-    elif 'version: 2.3.13' in version: pbs_params += "#PBS -l ncpus=%d\n"           % ncpus                  # blacklight
-    elif '14.2'            in version: pbs_params += "#PBS -l select=%d:ncpus=%d\n" % (nnodes, ppn)          # pbspro 14
-    elif is_cray                     : pbs_params += "#PBS -l size=%s\n"            % jd.total_cpu_count     # kraken, jaguar
-    else                             : pbs_params += "#PBS -l nodes=%d:ppn=%d%s\n"  % (nnodes, ppn,          # default
-                                       ''.join([':%s' % prop for prop in node_properties]))
+        pbs_params += '#PBS -l select=%d:ncpus=%d\n' % (nnodes, ppn)
```
Comment (andre-merzky) [inline src/radical/saga/adaptors/pbspro/pbsprojob.py:260]: This refactoring seems to have lost several version switches, if I am not mistaken (`'Version: 5.'` being a random example).  What is the rationale?  Cleanup of versions you would consider not be used anymore?  How would we know?

**Review by andre-merzky — APPROVED** [review]
Thank you for digging through this code and cleaning it up.  I left a couple of comments, but don't see major problems with the PR.

File: src/radical/saga/adaptors/pbspro/pbsprojob.py (hunk 4 above)
Comment (mtitov) [reply src/radical/saga/adaptors/pbspro/pbsprojob.py:257]: I cleaned out parts related to systems that were obsolete for awhile, and yes, considering that that versions are outdated. Regarding `'Version: 5.'` as I understood it is related mostly to Torque, which also uses `PBS` directives. So the original goal was to remove outdated-outdated ones

Comment (andre-merzky) [reply src/radical/saga/adaptors/pbspro/pbsprojob.py:258]: That's fair.  If push comes to shove, we have the versions in git history :-D

**Pull request merged**